### PR TITLE
Ensured upgraded groups are charge correctly

### PR DESCRIPTION
**Pull request by TheHollidayInn**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes put_issue_url_here

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

This ensures groups with existing members are charged correctly.

[//]: # (Put User ID in here - found in Settings -> API)

----
UUID:

### CODE REVIEW

File: website/server/libs/amazonPayments.js [hunk 1]
```diff
@@ -252,6 +252,13 @@ api.subscribe = async function subscribe (options) {
     if (!result) throw new NotAuthorized(i18n.t('invalidCoupon'));
   }
 
+  let amount = sub.price;
+
+  if (groupId) {
+    let group = await Group.findById(groupId).exec();
+    amount = sub.price + (group.memberCount - 1) * 3;
```
Comment (crookedneighbor) [inline website/server/libs/amazonPayments.js]: These look like magic numbers to me. Could we save these as variables so it's easier to reason what they mean?

**Review by crookedneighbor — DISMISSED** [review]

File: website/server/libs/stripePayments.js [hunk 2]
```diff
@@ -92,6 +92,8 @@ api.checkout = async function checkout (options, stripeInc) {
 
     if (groupId) {
       customerObject.quantity = sub.quantity;
+      let group = await Group.findById(groupId).exec();
```
Comment (paglias) [inline website/server/libs/stripePayments.js]: can we exclude the chat from the selected fields?

File: website/server/libs/amazonPayments.js (hunk 1 above)
Comment (paglias) [reply website/server/libs/amazonPayments.js]: agreed, a comment too would help

File: website/server/libs/amazonPayments.js [hunk 3]
```diff
@@ -252,6 +252,13 @@ api.subscribe = async function subscribe (options) {
     if (!result) throw new NotAuthorized(i18n.t('invalidCoupon'));
   }
 
+  let amount = sub.price;
+
+  if (groupId) {
+    let group = await Group.findById(groupId).exec();
```
Comment (paglias) [inline website/server/libs/amazonPayments.js]: can we exclude the chat from the selected fields?

**Review by paglias — DISMISSED** [review]

**Review by SabreCat — APPROVED** [review]

**Pull request merged**